src/components/Form/PasswordWidget.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+	FormControl,
+	IconButton,
+	InputAdornment,
+	OutlinedInput,
+	styled,
+} from '@mui/material';
+import * as React from 'react';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { WidgetProps } from '@rjsf/utils';
+import zxcvbn from 'zxcvbn';
+
+const STRENGTH_TITLES = ['Very weak', 'Weak', 'Good', 'Strong', 'Very strong'];
+const STRENGTH_STYLES = [
+	{ width: 0 },
+	{ width: '25%', backgroundColor: 'orange' },
+	{ width: '50%', backgroundColor: 'yellow' },
+	{ width: '75%', backgroundColor: 'green' },
+	{ width: '100%', backgroundColor: 'darkgreen' },
+];
+
+const PasswordStrengthContainer = styled('div')(({ theme }) => ({
+	paddingBottom: theme.spacing(1),
+	position: 'relative',
+	marginTop: theme.spacing(2),
+	marginBottom: theme.spacing(3),
+	fontSize: theme.typography.body2.fontSize,
+}));
+
+const PasswordStrengthMeter = styled('div')(() => ({
+	transition: 'all ease-out 150ms',
+	position: 'absolute',
+	height: '4px',
+	backgroundColor: 'red',
+	minWidth: '5px',
+	bottom: 0,
+}));
+
+const PasswordStrengthText = styled('div')(({ theme }) => ({
+	margin: 0,
+	marginBottom: theme.spacing(2),
+}));
+
+export interface PasswordStrengthProps {
+	password?: string;
+}
+
+const PasswordStrength = ({ password }: PasswordStrengthProps) => {
+	const [strengthScore, setStrengthScore] = React.useState<
+		number | undefined
+	>();
+
+	React.useEffect(() => {
+		if (zxcvbn && password) {
+			try {
+				const { score } = zxcvbn(password);
+				setStrengthScore(score);
+			} catch {
+				// Ignore any errors, as we only want to show the strength meter if it is available.
+			}
+		}
+	}, [password]);
+
+	if (!password || strengthScore === undefined) {
+		return null;
+	}
+
+	return (
+		<PasswordStrengthContainer>
```

Comment:
Perhaps we can use this and add `joy-ui` to our toolkit? https://mui.com/joy-ui/react-input/#password-meter

Reply:
for sure I can improve the code btw :) I'll push a better version

Reply:
Fair! And thanks for pushing the improved code :)